gateway/ln-gateway/proto/gatewaylnrpc.proto

```diff
@@ -38,12 +40,49 @@ message PayInvoiceResponse {
   bytes preimage = 1;
 }
 
-message SubscribeInterceptHtlcsRequest {
-  // Subscribe to intercepted HTLCs with this short channel id
-  uint64 short_channel_id = 1;
+message InterceptHtlcsRequest {
+  message Subscribe {
+    // The short channel id of HTLCs to intercept
+    uint64 short_channel_id = 1;
+  }
+
+  message Settle {
+    // The preimage for settling an intercepted HTLC
+    bytes preimage = 1;
+  }
+
+  message Cancel {
+    // The reason for the cancellation of an intercepted HTLC
+    string reason = 1;
```

Comment:
Maybe this should (also) be a LN error code we can send back to the sender?

Reply:
💯 , will formalize into standard LN error codes compatible/translatable across implementations